tensorflow/lite/core/model_builder.cc

```diff
@@ -90,6 +98,114 @@ std::unique_ptr<FlatBufferModel> FlatBufferModel::VerifyAndBuildFromBuffer(
                                       error_reporter);
 }
 
+void FlatBufferModel::ByteSwapSerializedModel(std::string* serialized_model){
+  const uint8_t* buffer =
+      reinterpret_cast<const uint8_t*>(serialized_model->c_str());
+  const tflite::Model* input_model = tflite::GetModel(buffer);
+  ByteSwapTFLiteModel(input_model);
+}
+
+void FlatBufferModel::ByteSwapBuffer(int8_t tensor_type, size_t buffer_size,
```

Comment:
Is the target always writable?
What happens if the target memory is mmaped ?

Reply:
Actually `FlatBufferModel::ByteSwapFlatBufferModel()` will allocate a new `FlatBufferModel` instance for byte swapping and return its handler.

Reply:
Then will it increase memory consumption?

Random thoughts, how about creating a separate tool to update LE buffers to BE buffers?

Reply:
Yes, it will increase memory consumption on BE platforms, but only during the model loading/saving phase, so I think it will not incur big impact to the performance.

Creating a separate tool will also work, but it will bring extra steps such as using the tool to convert the .tflite files on BE platforms before running TensorFlow code. I think embedding the byte swapping code in TensorFlow code base would be more convenient for the users, thus they can use the same steps and .tflite files across LE and BE platforms, without the need to mark which .tflite files are in which endianness format or do the conversion manually. The endianness issue becomes transparent to the users.